Tasks/AzureTestPlanV0/testLibExecutor.ts

```diff
@@ -0,0 +1,88 @@
+import os = require('os');
+import Q = require('q');
+import path = require('path');
+import fs = require('fs');
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as tr from 'azure-pipelines-task-lib/toolrunner';
+
+var isWindows = os.type().match(/^Win/);
+var m2HomeEnvVar: string = null;
+var mavenPOMFile: string = tl.getPathInput('mavenPOMFile', true, true);
+
+function getMavenExec(){
+    var mvnExec: string = '';
+    m2HomeEnvVar = tl.getVariable('M2_HOME');
+    if (m2HomeEnvVar) {
+        tl.debug('Using M2_HOME environment variable value for Maven path: ' + m2HomeEnvVar);
+        mvnExec = path.join(m2HomeEnvVar, 'bin', 'mvn');
+    }
+    // Second, look for Maven in the system path
+    else {
+        tl.debug('M2_HOME environment variable is not set, so Maven will be sought in the system path');
+        mvnExec = tl.which('mvn', true);
```

Comment:
Is it possible to not find the mvn in system path as well?. in that case do we need to handle by failing the task?

Reply:
If the mvn executor is not found in system path, we are failing the task by facing error while executing mvn -version first before trying to execute the tests. Please refer to execMavenBuild() function